tests/diffraction/powder/corrections_test.py

```diff
@@ -109,3 +111,157 @@ def test_merge_calibration_raises_if_mask_exists(calibration):
     )
     with pytest.raises(ValueError):
         merge_calibration(into=da, calibration=calibration)
+
+
+@pytest.mark.parametrize('data_dtype', ('float32', 'float64'))
+@pytest.mark.parametrize('dspacing_dtype', ('float32', 'float64'))
+@pytest.mark.parametrize('two_theta_dtype', ('float32', 'float64'))
+def test_lorentz_correction_dense_1d_coords(
+    data_dtype, dspacing_dtype, two_theta_dtype
+):
+    da = sc.DataArray(
+        sc.full(
+            value=2.1,
+            sizes={'detector_number': 3, 'dspacing': 4},
+            unit='counts',
+            dtype=data_dtype,
+        ),
+        coords={
+            'dspacing': sc.array(
+                dims=['dspacing'],
+                values=[0.1, 0.4, 0.7, 1.1],
+                unit='angstrom',
+                dtype=dspacing_dtype,
+            ),
+            'two_theta': sc.array(
+                dims=['detector_number'],
+                values=[0.8, 0.9, 1.3],
+                unit='rad',
+                dtype=two_theta_dtype,
+            ),
+            'detector_number': sc.array(
+                dims=['detector_number'], values=[0, 1, 2], unit=None
+            ),
+        },
+    )
+    original = da.copy(deep=True)
+    corrected = lorentz_correction(da)
+
+    assert corrected.sizes == {'detector_number': 3, 'dspacing': 4}
+    assert corrected.unit == 'angstrom**4 * counts'
+    assert corrected.dtype == original.dtype
+    assert not corrected.variances
+    assert not corrected.bins
+
+    d = original.coords['dspacing'].broadcast(sizes=corrected.sizes).values
+    two_theta = original.coords['two_theta'].broadcast(sizes=corrected.sizes).values
+    if any(dt == 'float32' for dt in (data_dtype, dspacing_dtype, two_theta_dtype)):
+        rtol = 1e-6
+    else:
+        rtol = 1e-15
+    np.testing.assert_allclose(
+        corrected.data.values, 2.1 * d**4 * np.sin(two_theta / 2), rtol=rtol
+    )
+
+    assert set(corrected.coords.keys()) == {'two_theta', 'dspacing', 'detector_number'}
+    for key, coord in corrected.coords.items():
+        sc.testing.assert_identical(coord, original.coords[key])
+
+
+def test_lorentz_factor_dense_2d_coord():
+    da = sc.DataArray(
+        sc.full(value=0.7, sizes={'detector_number': 3, 'dspacing': 4}),
+        coords={
+            'dspacing': sc.array(
+                dims=['dspacing'], values=[0.1, 0.4, 0.7, 1.1], unit='angstrom'
+            ).broadcast(sizes={'detector_number': 3, 'dspacing': 4}),
```

Comment:
Can we move this ``broadcast`` to line#195 instead?

Reply:
No. This test is specifically about testing with a 2d input coordinate.